refactor(AM-4304): Refactor code

gourishkb opened the pull request:
Refactored code to use methods, uses 'ctrl.Result, bool,error' pattern as much as possible.
Some functions like get<Object> which are repetative are also turned to methods
Build passes but, more refactoring is in progress

Note: Many method names are not ideal, would be glad for suggestions as well. 

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>

YachikaRalhan commented on controllers/serviceexport/reconciler.go:
```diff
@@ -62,18 +62,9 @@ var finalizerName = "networking.kubeslice.io/serviceexport-finalizer"
-		log.Error(err, "Failed to get serviceexport")
-		return ctrl.Result{}, err
+	serviceexport := r.GetServiceExport(ctx, req, &log)
+	if serviceexport == nil {
```
Shouldn't this return an actual error, if serviceexport not found, like it was returning previously

gourishkb replied:
You are right, I was trying to make it too terse. Fixed it

gourishkb commented:
Tests with iperf pass after addressing review comments, 
k exec -ti iperf-sleep-6ddf6f998c-dv29g -n iperf -c iperf -- iperf -c iperf-server.iperf.svc.slice.local -p 5201 -i 1 -b 10Mb;
------------------------------------------------------------
Client connecting to iperf-server.iperf.svc.slice.local, TCP port 5201
TCP window size: 45.0 KByte (default)
------------------------------------------------------------
[  1] local 192.168.1.5 port 58530 connected with 192.168.2.5 port 5201
[ ID] Interval       Transfer     Bandwidth
[  1] 0.00-1.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 1.00-2.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 2.00-3.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 3.00-4.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 4.00-5.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 5.00-6.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 6.00-7.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 7.00-8.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 8.00-9.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 9.00-10.00 sec  1.25 MBytes  10.5 Mbits/sec
[  1] 0.00-10.00 sec  12.9 MBytes  10.8 Mbits/sec
